### DISCUSSION TITLE
Variable number of parameters...

### DISCUSSION THREAD

**Discussion by gbernaldo82** [cesanta/mongoose#2466] [Ask questions, get answers] [answered]
Hi good afternoon!

I've been looking for some answer similar to this one but I haven't been able to find one. If someone has asked this before please just let me know about it.

We have a software written in C code which has to parse parameters in JSON format and the problem is that parameters are related to some commands and all the commands has a different number of parameters. Por example:

we have the path: **/api/settings**

and sometimes the content is
```
{
    "mixer":"0",    // Param 0
    "gain":"-30"    // Param 1
}
```

other times, the content is:

```
{
    "mixer":"0",                       // Param 0
    "upperThreshold":"-35", // Param 1
    "lowerThreashold":"-5"   // Param 2
}
```

With this, I mean the number of parameters of the json request is different depending on the command requested but the path is alwayse the same. Is not possible to create different paths for different commands because the software hadles hundreds if not thousands of commands. So my question is if there is any way to get the key and values and add them to a hash table data structure for example or something similar? 

Is it possible to do something like this?: 

```
struct mg_str json = hm->body;
for(int i = 0; i < mg_get_number_of_arguments(json); i++)
{
    char *pKey = mg_get_key(json, i);
    char *pValue = mg_get_value(json, i);
    HashTable_Add((void *)pKey, (void *)pValue);
}
```

Is this possible in mongoose?

Thank you so much in advance!

**Comment by scaprile** [comment, marked as answer]
> Is this possible in mongoose?

What's possible in Mongoose is what is described in its [documentation](https://mongoose.ws/documentation),  [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.

https://mongoose.ws/documentation/#json
For your particular case I guess https://mongoose.ws/documentation/#mg_json_next might work ?
https://github.com/cesanta/mongoose/blob/d18b2b390a7b6801349f1f62aa6d24ab67514fa6/test/unit_test.c#L2790-L2804

**Reply by gbernaldo82** [reply]
Yes, thank you so much for your answer.

This is what I was looking for!
Best regards!
